File: backend/src/main/java/com/bigdata/etl/model/entity/EtlTask.java
package com.bigdata.etl.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("etl_task")
public class EtlTask {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private Long sourceDatasourceId;
    private String sourceTable;
    private String targetTable;
    private String syncMode;       // FULL, INCREMENTAL
    private String incrementalColumn;  // for INCREMENTAL mode
    private String fieldMapping;   // JSON string: [{source, target}]
    private String scheduleType;   // MANUAL, CRON
    private String cronExpression;
    private String status;         // ENABLED, DISABLED
    private String lastRunStatus;  // SUCCESS, FAILED, RUNNING
    private LocalDateTime lastRunTime;
    private String remark;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/etlTask.ts
import request from './index'

export interface EtlTask {
  id?: number
  name: string
  sourceDatasourceId: number
  sourceTable: string
  targetTable: string
  syncMode: string
  incrementalColumn?: string
  fieldMapping?: string
  scheduleType: string
  cronExpression?: string
  status?: string
  remark?: string
}

export const etlTaskApi = {
  list: () => request.get('/etl/tasks'),
  create: (data: EtlTask) => request.post('/etl/tasks', data),
  update: (id: number, data: EtlTask) => request.put(`/etl/tasks/${id}`, data),
  delete: (id: number) => request.delete(`/etl/tasks/${id}`),
  run: (id: number) => request.post(`/etl/tasks/${id}/run`),
  logs: (id: number) => request.get(`/etl/tasks/${id}/logs`)
}
export interface EtlExecution {
  id?: number
  taskId: number
  status: string
  rowsProcessed: number
  errorMessage?: string
  startTime?: string
  endTime?: string
  createTime?: string
}

// Extend etlTaskApi with executions method
etlTaskApi.executions = (id: number) => request.get(`/etl/tasks/${id}/logs`)
etlTaskApi.listAll = () => request.get('/etl/tasks')
